lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -22,6 +22,49 @@
 
 LOG_MODULE_REGISTER(lte_lc_helpers, CONFIG_LTE_LINK_CONTROL_LOG_LEVEL);
 
+static int string_to_int(const char *str_buf, int base, int *output)
+{
+	int temp;
+	char *end_ptr;
+
+	errno = 0;
+	temp = strtol(str_buf, &end_ptr, base);
+
+	if (end_ptr == str_buf || *end_ptr != '\0' ||
+	    ((temp == LONG_MAX || temp == LONG_MIN) && errno == ERANGE))
+	{
+		return -ENODATA;
+	}
+
+	*output = temp;
+
+	return 0;
+}
+
+/* Converts integer on string format to integer type.
+ * Returns zero on success, otherwise negative error on failure.
+ */
+static int string_param_to_int(struct at_param_list *resp_list,
+			       size_t idx, int *output, int base)
+{
+	int err;
+	char str_buf[16];
```

Comment:
If I'm not mistaken this number could be increased to 33. 32 (+ 1 null character) If converting with base 2 to get to the maximum value of a unsigned 32-bit integer. But if exclusively using it for base 16 and 10 conversions 11 should be enough.

Reply:
I think I'll leave it as is for now, but do a follow-up PR to use this function in also other places in the helper file. Then the buffer size can be tuned to fit all use-cases